src/revChatGPT/WebGPT.py

```diff
@@ -0,0 +1,245 @@
+            !exit - Exit this program
+            !setconversation - Changes the conversation
+            """,
+            )
+        elif command == "!reset":
+            chatbot.reset_chat()
+            print("Chat session successfully reset.")
+        elif command == "!config":
+            print(json.dumps(chatbot.config, indent=4))
+        elif command.startswith("!rollback"):
+            try:
+                rollback = int(command.split(" ")[1])
+            except IndexError:
+                logging.exception(
+                    "No number specified, rolling back 1 message",
+                    stack_info=True,
+                )
+                rollback = 1
+            chatbot.rollback_conversation(rollback)
+            print(f"Rolled back {rollback} messages.")
+        elif command.startswith("!setconversation"):
+            try:
+                chatbot.conversation_id = chatbot.config[
+                    "conversation_id"
+                ] = command.split(" ")[1]
+                print("Conversation has been changed")
+            except IndexError:
+                log.exception(
+                    "Please include conversation UUID in command",
+                    stack_info=True,
+                )
+                print("Please include conversation UUID in command")
+        elif command == "!exit":
+            exit(0)
+        else:
+            return False
+        return True
+
+    session = V1.create_session()
+    completer = V1.create_completer(
+        ["!help", "!reset", "!config", "!rollback", "!exit", "!setconversation"],
+    )
+    print()
+    try:
+        while True:
+            print(V1.bcolors.OKBLUE + V1.bcolors.BOLD + "You:" + V1.bcolors.ENDC)
+
+            prompt = get_input(session=session, completer=completer)
+            if prompt.startswith("!"):
+                if handle_commands(prompt):
+                    continue
+
+            print()
+            print(V1.bcolors.OKGREEN + V1.bcolors.BOLD + "Chatbot: ")
+            prev_text = ""
+            for data in chatbot.webAsk(prompt):
+                message = data["message"][len(prev_text) :]
+                print(message, end="", flush=True)
+                prev_text = data["message"]
+            print(V1.bcolors.ENDC)
+            print()
+    except (KeyboardInterrupt, EOFError):
+        print("Exiting...")
+        exit(0)
+
+if __name__ == "__main__":
+    print(
+        """
+        WebGPT module. Copies functionality from WebChatGPT browser extension
+        Btw, the secret of the ddg-webapp-aagd.vercel.app is just duckduckgo_search.ddg() -Nymda
+        Repo: github.com/acheong08/ChatGPT
+        """,
+    )
+    print("Type '!help' to show a full list of commands")
+    print(
+        V1.bcolors.BOLD
+        + V1.bcolors.WARNING
+        + "Press Esc followed by Enter or Alt+Enter to send a message."
+        + V1.bcolors.ENDC,
+    )

```

Comment:
better to use `f'{V1.bcolors.BOLD} {V1.bcolors.WARNING} Press Esc followed by Enter or Alt+Enter to send a message. {V1.bcolors.ENDC}'`

Reply:
Ok, i just copy-pasted it from V1

Reply:
Yeah I also need to fix V1 lol